QamarDeen/QamarDeen/Services/NotificationScheduler.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  NotificationScheduler.swift
+//  QamarDeen
+//
+//  Created by Mazyad Alabduljaleel on 1/19/17.
+//  Copyright © 2017 Batoul Apps. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+/// responsible for wrapping the Local Notification APIs with
+/// convenient access to manage the required reminders
+final class NotificationScheduler {
+    
+    class var canScheduleAlerts: Bool {
+        
+        let app = UIApplication.shared
+        if let settings = app.currentUserNotificationSettings , settings.types != UIUserNotificationType() {
+            return true // settings enabled
+        }
+        
+        let types: UIUserNotificationType = [.sound, .alert]
+        let notifSettings = UIUserNotificationSettings(types: types, categories: nil)
+        
+        app.registerUserNotificationSettings(notifSettings)
```

Comment:
We should implement the AppDelegate method `application:didRegisterUserNotificationSettings:` so we can trigger off the user's granting access to then call scheduleNotifications.